lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescaleset_info.py

```diff
@@ -308,20 +305,20 @@ def exec_module(self, **kwargs):
         if self.name and not self.resource_group:
             self.fail("Parameter error: resource group required when filtering by name.")
 
+        if self.name:
+            result = self.get_item()
+        else:
+            result = self.list_items()
         if is_old_facts:
-            if self.name:
-                self.results['ansible_facts']['azure_vmss'] = self.get_item()
-            else:
-                self.results['ansible_facts']['azure_vmss'] = self.list_items()
+            self.results['ansible_facts'] = {
+                'azure_vmss': result
+            }
         else:
-            if self.name:
-                self.results['vmss'] = self.get_item()
-            else:
-                self.results['vmss'] = self.list_items()
+            self.results['vmss'] = result
 
         if self.format == 'curated':
```

Comment:
Shouldn't this go before the part above this where you are setting `self.results`. I'm struggling to see how this even affects the results in the facts module case.

Reply:
I didn't want to reorganize the code :) but yes, it makes more sense. `result` is a list, so even after assinging it to `self.results`, modifying it will affect the output.

I'll change the order.